### PR TITLE
Check that documents name CI mechanisms that actually run [#249]

### LINKED ISSUE #249: Check that documents naming a CI mechanism name one that actually runs
Three times now, a governed document has asserted in the present tense that a check runs, when it
did not. Each time it was found by accident, and each time the sentence sat in the very paragraph
written to protect the claim it got wrong.

## The three instances

| Where | Claimed | Reality | Found by |
|---|---|---|---|
| **ADR-005** | a governed-source lint rejecting `try`/`catch` in governed code | never written; `src/text/Raster.cpp` shipped in v0.5.0 with `try` and three `catch` clauses | #116, while writing the lint |
| **ADR-007 decision 6** | "the evidence tests run on … Clang, now that issue #48 re-enabled that leg" | #48 added the GCC 16 leg and left the Clang half of its own title open; `clang-build.yml` never carried a `push` trigger | #246, incidentally |
| **`clang-build.yml` header** | adding a Clang evidence step is future work, and the third-toolchain claim is still outstanding | the macOS lane had held it since #222 | #246, incidentally |

A fourth, of the same shape, in the same review round: the workflow step named **"Setup Clang 21"**
ran `update-alternatives` and printed `clang --version`, which reported **18.1.3** — twice — and
passed. The build genuinely used Clang 21, because the toolchain file addresses
`/usr/lib/llvm-21/bin` and never consults `PATH`. The step was evidence pointing at the wrong thing.

That is the pattern worth naming: **not a wrong fact, but a control that appears to be verified and
is not.** For a project whose regulatory argument is "verification is re-derivation, not
inspection", a green tick over zero assertions is the specific failure mode that matters.

## What I am *not* proposing

Linting prose. Whether a paragraph's claim is true is not decidable, and a checker that guesses will
be turned off within a month.

## What is actually checkable

A narrow, mechanical subset: when a document or a CI comment **names a mechanism**, resolve the name.

- **Workflow files** (`docs/**` and `.github/workflows/*` comments cite `*.yml`) — does the file
  exist, and does it have a `push`/`pull_request` trigger? A document saying a check "runs in CI"
  beside a `workflow_dispatch`-only workflow is exactly instance 2, and is decidable.
- **ctest labels** (`ctest -L <label>`) — does any test carry that label? This is what
  `--no-tests=error` guards at runtime; the same question can be asked statically.
- **CMake targets and tool binaries** (`mdux-*`) — is there an `add_executable`/`add_custom_target`
  of that name? This would have caught instance 1 at the moment ADR-005 was written.

A survey of the current tree suggests the backlog is small — the ADR claims I sampled resolve
correctly (ADR-003's compiler floors are really in `CMakeLists.txt:28-52`; ADR-007's float-specifier
lint is really `tools/evidence-lint/mdux_evidence_lint.py`). This is about keeping it that way, not
about a cleanup.

### The false-positive trap, since it decides whether this is usable

`docs/Vulkan-SC-Only-Migration-Plan.md:926` cites `.github/workflows/vulkansc-ci.yml`, which does not
exist — correctly, because it is inside a fenced code block proposing a file. A checker that flags
that is worse than none. It must skip fenced blocks, and it needs a way to mark a citation as
aspirational. Getting this wrong is how the check ends up disabled.

## Where it belongs

`tools/docs-lint/`, beside `check_schema_type_drift.py` and `check_file_headers.py` — same family,
same shape, same CI job, and the same convention of a `test_*.py` beside it.

## Acceptance

- [ ] A checker resolving named workflows, ctest labels and CMake/tool targets cited in `docs/**`
      and in `.github/workflows/**` comments.
- [ ] Fenced code blocks skipped, and an explicit opt-out for a deliberately aspirational citation,
      documented where an author will meet it.
- [ ] Unit tests beside it, matching the family's convention.
- [ ] Wired into the Documentation Lint job.
- [ ] The current tree passes without a suppression list. If it cannot, the exceptions are named in
      the issue rather than hidden in the tool.

**A note on scope.** The check cannot decide whether a claim is *true*, only whether the thing it
names *exists and is wired up*. That covers all four instances above, which is the argument for it —
but it should not be described, in this issue or in its own comments, as verifying more than that.
Doing so would reproduce the defect it exists to catch.

### PR DESCRIPTION
**Pull request by ambroise-leclerc**
## Summary

Add a standard-library-only documentation checker for the narrow subset of CI claims the tree can
resolve mechanically:

- local workflow citations must name an existing workflow with a `push` or `pull_request` trigger;
- `ctest -L` citations must name a label attached to at least one test;
- standalone `mdux-*` citations must resolve to a CMake target, host tool, lint script, or repository
  skill.

The checker scans Markdown prose under `docs/` and full-line workflow comments. It skips fenced
examples and supports a visible, same-line aspirational marker instead of a suppression list. The
failure paths, comment/fence boundaries, and the clean current tree are covered by unit tests, and
the checker now runs in Documentation Lint.

Closes #249

## Invitation and contributor agreement

- [x] Maintainer-authored change; an invitation is not applicable.
- [x] Maintainer-authored change; contributor CLA acceptance is not applicable.

Invitation / CLA issue: none — maintainer-authored

## Dependency

- **Base branch:** `develop`
- **Predecessor PR:** not stacked
- **Merge order:** mergeable now

## Shared registries touched

- [ ] `CMakeLists.txt` (root) — targets, trust-zone declarations, install/export set
- [ ] `FILE_SET CXX_MODULES` lists — a module interface added, removed or moved
- [ ] `tools/CMakeLists.txt` — host-tool targets and their sources
- [ ] `tests/CMakeLists.txt` — test source lists and suite membership
- [ ] Schemas under `docs/*/schemas/` or `docs/governance/schemas/`
- [ ] Generated indexes (`AI-Reference.{md,json}`, per-clause indexes)
- [ ] Committed artifacts under `generated/`
- [x] None of the above

## Rebase state

- **Rebased onto:** `develop` at `4c4413f2f0e6eda55a5e873cfc5e7a4676717a6d`
- **Conflicts resolved as a union:** no conflicts

## Verification

- [ ] `cmake --preset ninja-gcc && cmake --build build-gcc` — not run; this is host-only Python/documentation tooling with no C++ or CMake build-graph change
- [ ] `ctest --test-dir build-gcc` — not run for the same reason
- [x] `python3 tools/docs-lint/mdux_docs_lint.py` — 82 files checked, zero findings
- [x] Other:
  - `python3 -m unittest discover -s tools/docs-lint -p "test_*.py"` — 130 tests passed
  - `python3 tools/docs-lint/check_named_mechanisms.py` — 77 documents/workflows checked, zero findings or suppressions
  - `python3 tools/docs-lint/check_file_headers.py` — 195 sources checked
  - `python3 tools/docs-lint/generate_ai_reference.py --check` — all five indexes current
  - `python3 tools/docs-lint/check_schema_type_drift.py` — 7 schemas / 7 bindings / 2 vocabularies current
  - `git diff --check` — clean

## Post-merge gate

- [x] I understand that the `develop` workflow must be green after this merges before the next
      dependent PR may merge.
- **Post-merge `develop` run:** no successor
- [ ] That run is green.

## Regulatory impact

**Potentially safety-relevant.** This strengthens auditability by checking that named verification
mechanisms exist and are wired, but it does not change device runtime behavior, rendering, safety
controls, compliance metadata, or any claim of conformity with a medical-device standard.

Affected artifacts are the new host checker and unit tests, the Documentation Lint workflow, and
the author-facing convention in `CONTRIBUTING.md`. No regulatory corpus, requirement, hazard, risk
control, public interface, or generated evidence artifact changes. Residual uncertainty: the checker
intentionally does not decide whether surrounding prose is true; it verifies only the named
mechanism's existence and wiring described above.

## Summary by CodeRabbit

* **Documentation**
  * Added guidance for validating referenced workflows, test labels, and build targets.
  * Documented exemptions for aspirational references and fenced code blocks.

* **Quality Improvements**
  * Added automated validation to ensure named CI and build mechanisms resolve correctly.
  * Improved detection of missing triggers, labels, targets, malformed code fences, and invalid exemptions.

* **Tests**
  * Added comprehensive coverage for valid references, invalid references, and edge cases.

### CODE REVIEW

**Comment by ambroise-leclerc** [comment]
## Review findings fixed directly on this branch

Ten defects, all reproduced by running the checker before fixing and after. `3bd94eb`. CI green, including Documentation Lint; **36 tests pass**, the original 13 unchanged.

The structure held under probing — the three historical instances from #249 are caught, all fence forms are skipped, and a commented-out `push:` trigger does not fool it. What was loose was name recognition, in both directions.

### False positives on prose that is true

Each would have failed Documentation Lint on an ordinary sentence, which is how a linter gets switched off:

| Defect | Was |
|---|---|
| Sentence-ending period absorbed | `produced by mdux-shaderbake.` → target `mdux-shaderbake.` does not exist |
| Any bare `*.yml` read as a workflow | `dependabot.yml`, a file that exists, reported missing |
| `on: push` and `"on":` unrecognised | a workflow running on every push reported as having no trigger |
| Python checkers' diagnostic names unindexed | `mdux-file-headers` and `mdux-named-mechanisms` rejected |
| `-L` anywhere after `ctest` | `ctest(1) for -L semantics` invented a label |
| Exact label matching | `-L determin` called unattached, though `ctest -L` takes a regex |
| `LABELS unit fast` | only `unit` registered |

Bare workflow names are now only citations when they name a workflow; a `.github/workflows/` path is still checked, so nothing is lost. The index now reads the names the Python checkers **print**, so documenting this check no longer fails this check.

### The two blind spots

These mattered more, being the shape the checker exists to catch:

- An unclosed fence made every later line invisible with no diagnostic — now reported.
- `main()` printed `OK` after scanning zero files. A green tick over zero assertions, reachable by moving the script since the root is inferred from its own location. **Scanning nothing now fails.**

### One hardening worth your call

The aspirational marker now **requires** the tracking issue CONTRIBUTING already asked for, rather than CONTRIBUTING being softened. A bare marker is an untracked permanent exemption — the suppression list this deliberately does not have, spelled differently. No markers exist in the tree, so nothing breaks; say the word if you would rather relax the doc instead.

CONTRIBUTING also said to put the marker "on that comment". The checker exempts a *line*, so a marker on a comment block's last line would not have covered the citations above it. Now worded as same-line, matching both halves.

File: tools/docs-lint/check_named_mechanisms.py [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+            if indent == 0:
+                break
+            if child_indent is None:
+                child_indent = indent
+            if indent != child_indent:
+                continue
+            key = re.match(r"\s*(?P<key>[A-Za-z_][A-Za-z0-9_-]*)\s*:", child)
+            if key is not None and key.group("key") in AUTOMATIC_EVENTS:
+                return True
+        return False
```
[the remark this answers is not in the document]
Comment (ambroise-leclerc) [reply tools/docs-lint/check_named_mechanisms.py:144]: Fixed in `6b1b27b`. `has_automatic_trigger()` now recognizes direct block-sequence items, including quoted items and trailing comments, at the established child indentation. Regression coverage includes both `- push` and quoted `- pull_request` forms.

File: tools/docs-lint/test_check_named_mechanisms.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+            f"{finding.path.relative_to(root)}:{finding.line}: {finding.message}" for finding in findings
+        ]
+        self.assertEqual(rendered, [])
+
+
+if __name__ == "__main__":
+    unittest.main()
+
+
+class ReviewRegressionTests(CheckNamedMechanismsTests):
```
[the remark this answers is not in the document]
Comment (ambroise-leclerc) [reply tools/docs-lint/test_check_named_mechanisms.py]: Fixed in `6b1b27b`. The `unittest.main()` entry point now follows both test classes. Direct execution runs all 36 checker tests, matching discovery instead of exiting after the original class.

File: tools/docs-lint/test_check_named_mechanisms.py [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+    """One test per defect found reviewing this checker, so none of them can come back quietly."""
+
+    def test_a_citation_closing_a_sentence_keeps_its_full_stop_out_of_the_name(self) -> None:
+        """Exclude sentence punctuation from a bare tool name."""
+        with tempfile.TemporaryDirectory() as raw:
+            root = self.make_repository(raw)
+            self.assertEqual(self.findings_for(root, "Built by mdux-shaderbake.\n"), [])
```
[the remark this answers is not in the document]
Comment (ambroise-leclerc) [reply tools/docs-lint/test_check_named_mechanisms.py:205]: Fixed in `c708e2a`. The trailing boundary now permits sentence punctuation after a match while the captured mechanism still excludes the period. The regression covers both a known `mdux-shaderbake.` citation and an unknown `mdux-missing.` citation, which now reports `mdux-missing` as unresolved.

**Comment by ambroise-leclerc** [comment]
Addressed the two outside-diff findings from the CodeRabbit review summary in `1d082b8`:

- CTest label parsing now preserves the complete quoted regular expression, with regression coverage for `missing|determinism` selecting the attached `determinism` label.
- Workflow comment extraction now excludes comment-like content inside YAML `|`/`>` block scalars while continuing to scan genuine YAML comments after the scalar.

Verification: direct checker suite 36/36, full docs-lint discovery 153/153, and the current-tree mechanism scan is clean across 77 documents/workflows.

**Comment by ambroise-leclerc** [comment]
Addressed the two latest outside-diff findings in `97b407d`:

- `AUTOMATIC_EVENTS` now follows issue #249 exactly: only `push` and `pull_request` satisfy the checker. Regression coverage rejects `workflow_call` and `pull_request_target` as substitutes.
- CMake comment removal now blanks inline and bracket comments while preserving quoted/bracket arguments and source line structure. The regression proves an inline-commented `add_custom_target(mdux-inline-target)` is not indexed.

Verification remains clean: 36/36 focused tests, 153/153 docs-lint tests, `git diff --check`, and 77 current documents/workflows scanned without findings.

**Review by AM-L — APPROVED** [review]
